common/src/main/java/org/apache/falcon/security/DefaultAuthorizationProvider.java

```diff
@@ -170,7 +171,8 @@ public void authorizeResource(String resource, String action,
                     authorizeAdminResource(authenticatedUGI, action);
                 }
             } else if ("entities".equals(resource) || "instance".equals(resource)) {
-                authorizeEntityResource(authenticatedUGI, entityName, entityType, action);
+                authorizeEntityResource(authenticatedUGI, entityName, entityType, action,
+                        "entities".equals(resource) && LIST_OPERATION.equals(action));
```

Comment:
It is good practice to separate out functional logic from method calls. Have a boolean var that is set to "entities".equals(resource) && LIST_OPERATION.equals(action) and then use it in method call.